app/js/conceptSearch.controller.js

```diff
@@ -1,184 +1,185 @@
-
-        // Method used to prevent app from querying server with every letter input into search query panel
-        var timeout = null;
-        function timeoutRefresh()  {
-            clearTimeout(timeout);
-            vm.isUserTyping = true;
-            timeout = setTimeout(function () {
-                vm.refreshResponse();
-            }, 250);
-        }
-
-        // Method used to apply Rest response to controller variables and apply it on template
-        function refreshResponse() {
-            firstPage();
+    .controller('ConceptSearchController', ['$scope', '$routeParams', 'openmrsRest', '$translate',
+        function ($scope, $routeParams, openmrsRest, $translate) {
+
+            var vm = this;
+
+            $translate.refresh();
+
+            //Default is undefined, but it could take data from URL if there is one
+            vm.query;
+            vm.enableDescription = false;
+
+            //Tells if user is actually typing (which is described in timeoutRefresh() function
             vm.isUserTyping = false;
 
-            if (vm.query.length>0) {
-                openmrsRest.listFull('concept', {q: vm.query, limit: vm.entriesPerPage, includeAll: true}).then(function (firstResponse) {
-                    vm.loadingMorePages = true;
-                    vm.concepts = firstResponse;
-                    updateResultNotification();
+            //Default values
+            vm.entriesPerPage = 5;
+            vm.pageNumber = 1;
+            vm.concepts = '';
 
-                    openmrsRest.listFull('concept',{q: vm.query, includeAll: true}).then(function (response) {
-                        vm.concepts = response;
-                        updateResultNotification();
-                        vm.loadingMorePages = false;
-                    });
-                });
+            //Boolean that represents state of loading more pages after downloading first page
+            vm.loadingMorePages = false;
+
+            //Page navigation
+            vm.nextPage;
+            vm.prevPage;
+            vm.firstPage;
+            vm.lastPage;
+
+            //Page dividing
+            vm.sliceFrom;
+            vm.sliceTo;
+
+            //Page range
+            vm.viewRangeStart;
+            vm.viewRangeEnd;
+
+            // Method used to apply Rest response to controller variables and apply it on template
+            vm.refreshResponse;
+
+            //Method used to prevent app from querying server with every letter input into search query panel
+            vm.timeoutRefresh;
+
+            //Init method
+            activate();
+
+            //ng-show logic
+            function isSearching() {
+                return vm.isUserTyping && vm.query.length > 0
             }
-            else {
-                $scope.$apply(function () {
-                    vm.concepts='';
-                });
+
+            function isQueryFieldEmpty() {
+                return vm.query.length == 0 && !vm.isUserTyping;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2321639275/issues/source?bid=3196969&fileBranchId=3199483#l54)